src/Swarm/TUI/Controller.hs

```diff
@@ -399,9 +399,14 @@ saveScenarioInfoOnQuit = do
           Nothing -> return ()
           Just si -> liftIO $ saveScenarioInfo p si
 
-        -- rebuild the NewGameMenu so it gets the updated ScenarioInfo
+        -- See what scenario is currently focused in the menu.  Depending on how the
+        -- previous scenario ended (via quit vs. via win), it might be the same as
+        -- currentScenarioPath or it might be different.
+        curPath <- preuse $ uiState . uiMenu . _NewGameMenu . neHead . BL.listSelectedElementL . _SISingle . _2 . scenarioPath
```

Comment:
Could we use [`first1Of`](https://hackage.haskell.org/package/lens-5.2/docs/Control-Lens-Combinators.html#v:first1Of) or [`_head`](https://hackage.haskell.org/package/lens-5.2/docs/Control-Lens-Combinators.html#v:_head) or `ix 0` since you only need a traversal?

This seems like such a common thing that I would expect the `lens` package to provide a safe function to access the first element of a non-empty list. :thinking:

Reply:
I definitely tried using `_head` before, but there is no `Cons` instance for `NonEmpty`.  I hadn't thought of doing `ix 0`, I will give that a try.